src/components/interactive-builder/modals/question/question-form/rendering-types/inputs/workspace-launcher/workspace-launcher.test.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import WorkspaceLauncher from './workspace-launcher.component';
+import { FormFieldProvider } from '../../../../form-field-context';
+import type { FormField } from '@openmrs/esm-form-engine-lib';
+
+const mockSetFormField = jest.fn();
+const formField: FormField = {
+  type: 'obs',
+  questionOptions: {
+    rendering: 'workspace-launcher' as const,
+    buttonLabel: '',
+    workspaceName: '',
+  },
+  id: '1',
+};
+
+jest.mock('../../../../form-field-context', () => ({
+  ...jest.requireActual('../../../../form-field-context'),
+  useFormField: () => ({ formField, setFormField: mockSetFormField }),
+}));
+
+describe('WorkspaceLauncher', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders workspace launcher inputs', () => {
+    renderWorkspaceLauncher();
+    expect(screen.getByLabelText(/Button Label/i)).toBeInTheDocument();
+    expect(screen.getByLabelText(/Workspace Name/i)).toBeInTheDocument();
+  });
+
+  it('displays placeholder text for both inputs', () => {
+    renderWorkspaceLauncher();
+    expect(screen.getByPlaceholderText(/Enter text to display on the button/i)).toBeInTheDocument();
+    expect(screen.getByPlaceholderText(/Enter the name of the workspace to launch/i)).toBeInTheDocument();
+  });
+
+  it('marks both inputs as required', () => {
+    renderWorkspaceLauncher();
+    const buttonLabelInput = screen.getByLabelText(/Button Label/i);
+    const workspaceNameInput = screen.getByLabelText(/Workspace Name/i);
+
+    expect(buttonLabelInput).toBeRequired();
+    expect(workspaceNameInput).toBeRequired();
+  });
+
+  it('updates button label when input changes', () => {
+    renderWorkspaceLauncher();
+    const buttonLabelInput = screen.getByLabelText(/Button Label/i);
+    const newValue = 'Launch Patient Dashboard';
+
+    fireEvent.change(buttonLabelInput, { target: { value: newValue } });
```

Comment:
Prefer to use userEvent over fireEvent - https://testing-library.com/docs/user-event/intro#differences-from-fireevent